modules/openthread/platform/diag.c

```diff
@@ -291,25 +319,152 @@ otError otPlatDiagGpioGetMode(uint32_t aGpio, otGpioMode *aMode)
+{
+	uint32_t now;
+	otRadioFrame *sTxPacket;
+	const uint16_t diag_packet_len = 30;
+
+	if (sTransmitMode == diag_transmit_mode_packets) {
+		if ((sTxCount > 0) || (sTxCount == -1)) {
+			sTxPacket = otPlatRadioGetTransmitBuffer(aInstance);
+
+			sTxPacket->mInfo.mTxInfo.mTxDelayBaseTime = 0;
+			sTxPacket->mInfo.mTxInfo.mTxDelay = 0;
+			sTxPacket->mInfo.mTxInfo.mMaxCsmaBackoffs = 0;
+			sTxPacket->mInfo.mTxInfo.mMaxFrameRetries = 0;
+			sTxPacket->mInfo.mTxInfo.mRxChannelAfterTxDone = sChannel;
+			sTxPacket->mInfo.mTxInfo.mTxPower = OT_RADIO_POWER_INVALID;
+			sTxPacket->mInfo.mTxInfo.mIsHeaderUpdated = false;
+			sTxPacket->mInfo.mTxInfo.mIsARetx = false;
+			sTxPacket->mInfo.mTxInfo.mCsmaCaEnabled = false;
+			sTxPacket->mInfo.mTxInfo.mCslPresent = false;
+			sTxPacket->mInfo.mTxInfo.mIsSecurityProcessed = false;
+
+			sTxPacket->mLength = diag_packet_len;
+
+			for (uint8_t i = 0; i < diag_packet_len; i++) {
+				sTxPacket->mPsdu[i] = i;
+			}
+
+			otPlatRadioTransmit(aInstance, sTxPacket);
+
+			if (sTxCount != -1) {
+				sTxCount--;
+			}
+
+			now = otPlatAlarmMilliGetNow();
+			otPlatAlarmMilliStartAt(aInstance, now, sTxPeriod);
+		} else {
+			sTransmitMode = diag_transmit_mode_idle;
+			otPlatAlarmMilliStop(aInstance);
+			otPlatLog(OT_LOG_LEVEL_DEBG, OT_LOG_REGION_PLATFORM, "Transmit done");
+		}
+	}
+}
+
+static otError processTransmit(otInstance *aInstance, uint8_t aArgsLength, char *aArgs[])
+{
+	otError error = OT_ERROR_NONE;
+	long value;
+	uint32_t now;
+
+	if (aArgsLength == 0) {
+		diag_output("transmit will send %" PRId32 " diagnostic messages with %" PRIu32
+			    " ms interval\r\n",
+			    sTxRequestedCount, sTxPeriod);
+
+	} else if (strcmp(aArgs[0], "stop") == 0) {
+		if (sTransmitMode == diag_transmit_mode_idle) {
+			return OT_ERROR_INVALID_STATE;
+		}
+
+		otPlatAlarmMilliStop(aInstance);
+		diag_output("diagnostic message transmission is stopped\r\nstatus 0x%02x\r\n",
+			    error);
+		sTransmitMode = diag_transmit_mode_idle;
+		otPlatRadioReceive(aInstance, sChannel);
+
+	} else if (strcmp(aArgs[0], "start") == 0) {
+		if (sTransmitMode != diag_transmit_mode_idle) {
+			return OT_ERROR_INVALID_STATE;
+		}
+
+		otPlatAlarmMilliStop(aInstance);
+		sTransmitMode = diag_transmit_mode_packets;
+		sTxCount = sTxRequestedCount;
+		now = otPlatAlarmMilliGetNow();
+		otPlatAlarmMilliStartAt(aInstance, now, sTxPeriod);
+		diag_output("sending %" PRId32 " diagnostic messages with %" PRIu32
+			    " ms interval\r\nstatus 0x%02x\r\n",
+			    sTxRequestedCount, sTxPeriod, error);
+	} else if (strcmp(aArgs[0], "interval") == 0) {
+
+		if (aArgsLength != 2) {
+			return OT_ERROR_INVALID_ARGS;
+		}
+
+		error = parse_long(aArgs[1], &value);
+		if (error != OT_ERROR_NONE) {
+			return error;
+		}
+
+		if (value <= 0) {
+			return OT_ERROR_INVALID_ARGS;
+		}
+		sTxPeriod = (uint32_t)(value);
+		diag_output("set diagnostic messages interval to %" PRIu32
+			    " ms\r\nstatus 0x%02x\r\n",
+			    sTxPeriod, error);
```

Comment:
```suggestion
		diag_output("set diagnostic messages interval to %" PRIu32
			    " ms\r\n", sTxPeriod);
```
we get here only if `error==OT_ERROR_NONE`